src/Infrastructure.Windows/SoftwareManagers/OneGameLauncherManager.cs

```diff
@@ -30,9 +31,28 @@ IProcessService processService
 
     public string DefaultArguments => "shell:appsfolder\\62269AlexShats.OneGameLauncherBeta_gghb1w55myjr2!App";
 
-    public bool IsRunning(string softwarePath) => this.User32Service.FindWindow("ApplicationFrameWindow", "One Game Launcher (Free)") != IntPtr.Zero;
+    private nint Window => this.User32Service.FindWindow("ApplicationFrameWindow", "One Game Launcher (Free)");
 
-    public void Start(string softwarePath, string? softwareArguments) => this.ProcessService.Start(softwarePath, softwareArguments).Dispose();
+    public void MaximizeWindow()
+    {
+        if (this.User32Service is WindowsUser32Service s && Window != IntPtr.Zero)
+        {
+            s.SetForegroundWindow(Window);
+            s.ShowWindowAsync(Window, 3);
+        }
+    }
+
+    public bool IsRunning(string softwarePath)
+    {
+        MaximizeWindow();
+        return Window != IntPtr.Zero;
+    }
```

Comment:
Should I keep the `MaximizeWindow()` function or remove it?

Reply:
no strange behavior, I just found it interesting to keep it because normally I don't use the maximized launcher when I'm on the PC, but when I access it via parsec through the retroid, I prefer it maximized because the retroid has a very low resolution (640x480) compared to my monitor (2160x1080)

Reply:
Oh maximized! 🤦‍♂️My mistake! I had mine maximized the whole time lol. Yeah, let's leave it in. Just add the `ShowWindowAsync` function to the `IUser32Service` interface to avoid the cast to `WindowsUser32Service`, and maybe store the value of `this.Window` in a local variable so `this.User32Service.FindWindow` is only called once from `MaximizeWindow`.

Thanks for pointing that out!

Reply:
Window is currently used in `MaximizeWindow()` and IsRunning. Can I keep this property and just switch it to `private`?

```csharp
private nint Window => this.User32Service.FindWindow("ApplicationFrameWindow", "One Game Launcher (Free)");    

    public bool IsRunning(string softwarePath) => Window == this.User32Service.GetForegroundWindow();

    public void Start(string softwarePath, string? softwareArguments)
    {
        ProcessService.Start(softwarePath, softwareArguments).Dispose();
        this.User32Service.SetForegroundWindow(Window);
        this.User32Service.ShowWindowAsync(Window, 3);
    }
```

I dropped the `MaximizeWindow()` function since it will be executed only when starting the process.

Reply:
The property is good. This is what I was talking about:

```csharp
IntPtr w = this.Window;
if (w != IntPtr.Zero)
{
    this.User32Service.SetForegroundWindow(w);
    this.User32Service.ShowWindowAsync(w, 3);
}
```

It avoids calling the Window property getter more than once, which avoids calling the `this.User32Service.FindWindow` function more than once. Not a huge deal.